src/ruby/yast/wfm.rb

```diff
@@ -263,5 +266,13 @@ def self.run_client(client)
         return false
       end
     end
+
+    private_class_method def self.ensure_path(path)
+      return Path.new(path) if path.is_a?(::String)
+      return path if path.is_a?(Yast::Path)
+
+      raise ArgumentError, "argument '#{path.inspect}' is not path or string"
```

Comment:
Uh, again? Repeating just twice is OK...

but here's a suggestion to save this repetition: move replace `ensure_path` to `Path`, perhaps even to `Path.new` ?
https://github.com/yast/yast-ruby-bindings/blob/87cb0136852c976ac72659504f294ffb07d45aaa/src/ruby/yast/path.rb#L10

```rb
    def initialize(value)
      return value.clone if value.is_a?(Yast::Path)
      if !path.is_a?(::String)
        raise ArgumentError, "Argument #{value.inspect} is neither a ::String or a Yast::Path"
      end

      @components = []
      load_components value
    end
```

AFAIK Yast::Path is not checking that it gets a String, and adding that could be useful(?)